src/middle_end/matchcomp.ml

```diff
@@ -257,20 +280,9 @@ let matrix_head_constructors mtx =
     | (p::_, mb)::tl -> help tl (pattern_head_constructors_aux p acc) in
   help mtx []
 
-let normalize_branches match_branches =
-  let match_branches = List.map normalize_branch match_branches in
-  let rec help branches =
-    match branches with
-    | [] -> []
-    | hd :: _ when pattern_always_matches hd.mb_pat -> [hd]
-    | hd :: tl -> hd :: (help tl) in
-  help match_branches
-
 let make_matrix branches =
-  List.map (fun ({mb_pat} as mb) ->
-      match mb_pat.pat_desc with
-      | TPatTuple(patts) -> patts, mb
-      | _ -> ([mb_pat], mb)) branches
+  (* Form matrix and assign each branch a number *)
```

Comment:
This was done to make debugging much easier. The switch case numbers in the compiled output now map 1:1 onto the branches in the original program.